tests/SpecLazy.elm

```diff
@@ -1,188 +0,0 @@
-module SpecLazy exposing (suite)
```

Comment:
This seemed like it was just copy/pasted over from `tests/Spec` which is why I deleted it... but might be useful since Lazy is back? If this is just testing to verify the Lazy versions still work fine, what I'm curious about is if we can't actually test the laziness of the versions in the Lazy module more specifically somehow?